common/comms/queue_adaptor.py

```diff
@@ -0,0 +1,20 @@
+"""Module for generic queue adaptor functionality"""
+import abc
+
+
+class QueueAdaptor(abc.ABC):
+    """Interface for a message queue adaptor."""
+
+    async def send_async(self, message: str) -> None:
+        """
+        Sends a message with awaits using the async flow.
+        :param message: The message content to send.
+        """
+        pass
+
+    def send_sync(self, message: str) -> None:
```

Comment:
I don't see why this is part of the interface, when would we ever use it?

Reply:
I exposed this as part of the interface in cases where an alternative means than using a tornado executor to cause the sync functionality to be run async was desired.